src/Control/RequestHandler.php

```diff
@@ -178,7 +178,7 @@ public function handleRequest(HTTPRequest $request)
 
         // Actions can reference URL parameters, eg, '$Action/$ID/$OtherID' => '$Action',
         if ($action[0] == '$') {
-            $action = str_replace("-", "_", $request->latestParam(substr($action, 1)));
+            $action = str_replace("-", "_", $request->latestParam(substr($action, 1)) ?: '');
```

Comment:
A lot of string methods will now throw warnings when you give them a `null` value. It's a bit disconcerting how many places we actually end up with `null` when the method expects a string. In some places, we even go out of our way to make sure a `null` value can be provided (e.g.: HTTPRequest body).

I've mostly fixed this by sticking a `?: ''` after the nullable value. Although part of me feels the proper way to fix this is to update the logic so null values never end up in a method that expects a string.

Reply:
> part of me feels the proper way to fix this is to update the logic so null values never end up in a method that expects a string

I agree with this in principle, but it's done so much in Silverstripe that it's become reasonably standard to return null instead of an empty string in project code. Anything done to ensure null values don't end up where strings are expected will need to be careful not to break project code that is returning nulls.

Reply:
rector took the approach of using `(string) $str` - https://github.com/silverstripe/silverstripe-framework/pull/10222

The cast approach is probably a bit better than `?: ''` as there's a change that we'd end up with a function that returns something like '1' or 'true' and causes a type error